src/Models/ModelInstantiator.php

```diff
@@ -83,6 +82,12 @@ public function arrayToDataObject(
                 $instance->{$property} = $instances;
             } else {
                 // ManyToOne: Map to a single nested Data Object.
+
+                if (is_null($arr[$property]) && $this->propertyAllowsNull($property, $property_reflection)) {
```

Comment:
It may make sense to use `$enforcer->propIsNull()` here: https://github.com/PDGA/php-data-objects/blob/9f8eb4ee428df91c9c516a6d5d67492be6125a29/src/Enforcers/ValidationEnforcer.php#L147

That would add another call to `propIsUndefined`; not sure if it's worth the tradeoff.